### PR TITLE
deps: remove dependency to `pedantic`

### LINKED ISSUE #8: Request : remove dependency to `pedantic`
The `pedantic` library is considered **deprecated** : see https://github.com/google/pedantic#readme .

There are migration instructions here : https://github.com/dart-lang/lints#migrating-from-packagepedantic .

`unawaited` is now available natively as part of `dart:async` so we could maybe rely on it instead, and remove the `pedantic` dependency all together ?

Thanks !

### PR DESCRIPTION
**Pull request by tsimbalar**
`pedantic` is deprecated, and has been superseeded by `flutter_lints`

See also : 
- Introducing `flutter_lints`  https://docs.flutter.dev/release/breaking-changes/flutter-lints-package
- Migration instructions :  https://github.com/dart-lang/lints#migrating-from-packagepedantic

I thought this library actually relied on `unawaited` from `pedantic` , but it doesn't ... still migrated to recommended linters, which hopefully will also help increase the score on `pub.dev`

Cheers !

fixes #8

### CODE REVIEW

File: example/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+include: package:flutter_lints/flutter.yaml
```
Comment (tsimbalar) [inline example/analysis_options.yaml:1]: Had to add also a `analysis_options.yaml` under `example` because VSCode was otherwise complaining, and using the file from the root of the repo

File: example/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+linter:
+  # in the example project, we are a bit less strict
+  rules:
+    use_key_in_widget_constructors: false
+    prefer_const_constructors_in_immutables: false
+    prefer_const_constructors: false
+    prefer_const_literals_to_create_immutables: false
```
Comment (tsimbalar) [inline example/analysis_options.yaml:9]: disabled the rules that were triggered in the `example` project. 

Seems like we probably don't care that much about following the guidelines in the example project because it's not really shared code.

File: pubspec.lock [hunk 3]
```diff
@@ -55,39 +55,46 @@ packages:
...
       url: "https://pub.dartlang.org"
     source: hosted
     version: "1.8.0"
-  pedantic:
-    dependency: "direct main"
-    description:
-      name: pedantic
-      url: "https://pub.dartlang.org"
-    source: hosted
-    version: "1.11.0"
```
Comment (tsimbalar) [inline pubspec.lock:90]: this was the goal

File: test/horizontal_blocked_scroll_physics_test.dart [hunk 4]
```diff
@@ -105,25 +105,25 @@ MovementResult moveRightNotInLeftRange(HorizontalBlockedScrollPhysics hs) {
 void main() {
   group('HorizontalBlockScrollPhysics', () {
     test('won\'t block by default', () {
-      var hs = HorizontalBlockedScrollPhysics();
+      const hs = HorizontalBlockedScrollPhysics();
```
Comment (tsimbalar) [inline test/horizontal_blocked_scroll_physics_test.dart:108]: applying recommendations from `flutter_lints` so `flutter analyze` does not complain

**Comment by tsimbalar** [comment]
@robertohuertasm do I need to increment the version number in `pubspec` (and `CHANGELOG`) ? or do you take care of that ?

**Review by robertohuertasm — APPROVED** [review]
LGTM 🚀  Feel free to change the version and the changelog. You can also take care of the publication if you want. Just create a release and it should work. Thank you for the PR ❤️

**Comment by tsimbalar** [comment]
@robertohuertasm I've bumped version / updated changelog / merged and created a release ... 
but the package publication seems to be failing 😬 

https://github.com/koa-health/horizontal_blocked_scroll_physics/runs/5444959607?check_suite_focus=true

```
Uploading...
OAuth2 authorization failed (Bad Request).
Pub needs your authorization to upload packages on your behalf.
In a web browser, go to https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force&response_type=code&client_id=818368855108-8grd2eg9tj9f38os6f1urbcvsq399u8n.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A33595&code_challenge=goUFxyjCO61sDyqg8-GbXXj0lSzert5zeAXVsKuBxu8&code_challenge_method=S256&scope=openid+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email
Then click "Allow access".
Waiting for your authorization...
```

using the link doesn't seem to work for me (trying to access localhost 🤔 ?) 

image

**Comment by tsimbalar** [comment]
ok, I ran `flutter pub publish` from my computer directly ... I was expecting a auth error or something there ... but it worked .

TL;DR : it's published 😬 
https://pub.dev/packages/horizontal_blocked_scroll_physics/versions/4.0.1

**Comment by robertohuertasm** [comment]
That's probably because the auth token may be expired.
